### PR TITLE
Issues/#233 enable some tests + update genesis block

### PR DESCRIPTION
**Pull request by r-soltani**
This PR reenabled a number of tests only. 
The purpose is to update the genesis block with new hashs based on temporary difficulty(nbit)  of 0x1f00ffff.

### CODE REVIEW

File: src/test/base58_tests.cpp [hunk 1]
```diff
@@ -5,8 +5,16 @@
 #include <base58.h>
 
 #include <test/data/base58_encode_decode.json.h>
+#ifdef BUILD_BTC
+#include <test/data/base58_keys_invalid.json.h>
+#else // BUILD_EQB
 #include <test/data/base58_keys_invalid.json.h>
```
Comment (macterra) [inline src/test/base58_tests.cpp]: Shouldn't we be including the eqb_ versions here?

Comment (r-soltani) [reply src/test/base58_tests.cpp]: No, next PR will address that by updating the content

Comment (macterra) [reply src/test/base58_tests.cpp]: :-/

**Review by macterra — APPROVED** [review]
Approved assuming pending changes in next PR

**Review by ealefad — APPROVED** [review]

File: src/amount.h [hunk 2]
```diff
@@ -29,7 +29,7 @@ static const CAmount CENT = 1000000;
 static const CAmount MAX_MONEY = 21000000 * COIN;
 #else  // BUILD_EQB
 static const CAmount MAX_MONEY              = 22000000  * COIN;
-static const CAmount GENESIS_BLOCK_REWARD   =       50  * COIN; // EQB_TODO: Keeping reward same as Bitcoin to pass tests. It is unspendable in Bitcoin.
+static const CAmount GENESIS_BLOCK_REWARD   =        0  * COIN;
```
Comment (KiarashN) [inline src/amount.h]: Would you replace "0 * COIN" by "0"

Comment (r-soltani) [reply src/amount.h]: Done

File: src/test/base58_tests.cpp (hunk 1 above)
Comment (r-soltani) [reply src/test/base58_tests.cpp]: I added this fix to this PR as well, along with another change

**Pull request merged**